hedera-node/hedera-file-service-impl/src/main/java/com/hedera/node/app/service/file/impl/utils/FileServiceUtils.java

```diff
@@ -77,19 +86,85 @@ public static void validateContent(@NonNull byte[] content, @NonNull FilesConfig
 
     /**
      * The function validates the keys and adds them to the context.
+     *
      * @param listKeys the list of keys to validate and add to required keys in context
      * @param context the prehandle context for the transaction.
-     * @param areKeysRequired create allows files to be created without additional keys. Therefore, this flag is needed.
+     * @param areKeysRequired create allows files to be created without additional keys. Therefore,
+     *     this flag is needed.
      * @throws PreCheckException
      */
     public static void validateAndAddRequiredKeys(
-            @Nullable final KeyList listKeys, @NonNull final PreHandleContext context, final boolean areKeysRequired)
+            @Nullable final KeyList listKeys,
+            @NonNull final PreHandleContext context,
+            final boolean areKeysRequired)
             throws PreCheckException {
         if (listKeys == null || !listKeys.hasKeys() || listKeys.keys().isEmpty()) {
             // @todo('protobuf change needed') change to immutable file response code
             if (areKeysRequired) throw new PreCheckException(UNAUTHORIZED);
         }
 
-        if (listKeys != null && listKeys.hasKeys()) for (Key key : listKeys.keys()) context.requireKey(key);
+        if (listKeys != null && listKeys.hasKeys())
+            for (Key key : listKeys.keys()) context.requireKey(key);
+    }
+
+    /**
+     * The function validates the file id and returns the file metadata for system files.
+     *
+     * @param handleContext the handle context for the transaction.
+     * @param fileStore the file store to fetch the metadata of specified file id
+     * @param fileId the file id to validate and to fetch the metadata
+     * @param canBeDeleted flag if the file can be deleted or not
```

Comment:
`flag indicating that the file may be deleted, and the call should not fail if it is.`

Reply:
rewrite it thanks